main/com.tunein/manifest.tt

```diff
@@ -0,0 +1,139 @@
+                                             out now_playing_id: String 
+                                             #[string_values="com.tunein:now_playing_id"], 
+                                             out URL: Entity(tt:url)
+                                             out formats: String 
+                                             #[string_values="com.tunein:formats"],
+                                             image: Entity(tt:picture))
+  #_[canonical="list local radio channels"]
+  #_[confirmation="Here are the available channels in your area"]
+  #_[formatted=[{type="rdl",webCallback="${URL},displayTitle="${title}",displayText="${description}"}, {type="picture",url="${image}"}]]
+  #[poll_interval=100000ms]
+  #_[on_error={streaming_service_unaccessible=["radio streaming service is currently unaccessible"]}]
+  #[doc="list all the available radio channels in your area"];
+
+  monitorable query list list_trending_channels(in req query: String
+                                                #[string_values="tt:search_query"],
+                                                in opt count: Number 
+                                                #_[prompt="How many search results do you want?"],
+                                                out text: String 
+                                                #[string_values="com.tunein:text"],
+                                                out subtext: String 
+                                                #[string_values="com.tunein:subtext"],
+                                                out current_track: String 
+                                                #[string_values="com.tunein:current_track"],
+                                                out now_playing_id: String 
+                                                #[string_values="com.tunein:now_playing_id"], 
+                                                out URL: Entity(tt:url)
+                                                out formats: String 
+                                                #[string_values="com.tunein:formats"],
+                                                image: Entity(tt:picture))
+  #_[canonical="list trending radio channels"]
+  #_[confirmation="Here are the trending channels"]
+  #_[formatted=[{type="rdl",webCallback="${URL},displayTitle="${title}",displayText="${description}"}, {type="picture",url="${image}"}]]
+  #[poll_interval=100000ms]
+  #_[on_error={streaming_service_unaccessible=["radio streaming service is currently unaccessible"]}]
+  #[doc="list the trending radio channels"];
+
+  action play(in req query: String
+              #[string_values="tt:search_query"]
+              #_[prompt=["which channel would you like to listen to?"]]
+              #_[canonical={
+                 base=["text"],
+                 property=["# radio", "# channel"]}],
+              out URL: Entity(tt:url))
```

Comment:
`play` should take a single parameter of entity type, the ID of the radio station to listen to. That way, you get the right type-based composition between search and play.
(Look at spotify for how it's done)